lib/concurrent/configuration.rb

```diff
@@ -1,124 +1,296 @@
+  #   When this is set to true (the default) then `at_exit` handlers
+  #   will be registered automatically for the *global* thread pools
+  #   to ensure that they are shutdown when the application ends. When
+  #   changed to false, the `at_exit` handlers will be circumvented
+  #   for all *global* thread pools. This method should *never* be called
+  #   from within a gem. It should *only* be used from within the main
+  #   application and even then it should be used only when necessary.
+  #
+  def self.disable_auto_termination_of_global_executors!
+    @@auto_terminate_global_executors.make_false
+  end
 
-    # defines if executors should be auto-terminated in at_exit callback
-    attr_accessor :auto_terminate
+  # Defines if global executors should be auto-terminated with an
+  # `at_exit` callback. When set to `false` it will be the application
+  # programmer's responsibility to ensure that the global thread pools
+  # are shutdown properly prior to application exit.
+  #
+  # @note Only change this option if you know what you are doing!
+  #   When this is set to true (the default) then `at_exit` handlers
+  #   will be registered automatically for the *global* thread pools
+  #   to ensure that they are shutdown when the application ends. When
+  #   changed to false, the `at_exit` handlers will be circumvented
+  #   for all *global* thread pools. This method should *never* be called
+  #   from within a gem. It should *only* be used from within the main
+  #   application and even then it should be used only when necessary.
+  #
+  # @return [Boolean] true when global thread pools will auto-terminate on
+  #   application exit using an `at_exit` handler; false when no auto-termination
+  #   will occur.
+  def self.auto_terminate_global_executors?
+    @@auto_terminate_global_executors.value
+  end
+
+  # Defines if *ALL* executors should be auto-terminated with an
+  # `at_exit` callback. When set to `false` it will be the application
+  # programmer's responsibility to ensure that *all* thread pools,
+  # including the global thread pools, are shutdown properly prior to
+  # application exit.
+  #
+  # @note Only change this option if you know what you are doing!
+  #   When this is set to true (the default) then `at_exit` handlers
+  #   will be registered automatically for *all* thread pools to
+  #   ensure that they are shutdown when the application ends. When
+  #   changed to false, the `at_exit` handlers will be circumvented
+  #   for *all* Concurrent Ruby thread pools running within the
+  #   application. Even those created within other gems used by the
+  #   application. This method should *never* be called from within a
+  #   gem. It should *only* be used from within the main application.
+  #   And even then it should be used only when necessary.
+  def self.disable_auto_termination_of_all_executors!
+    @@auto_terminate_all_executors.make_false
+  end
```

Comment:
Missing on purpose. It's on by default and *most* users will probably not have the need/knowledge to change it. Once an advanced user turns it off, I didn't see any reason they would need to turn it back on. So I considered the counterpart method to be YAGNI.